src/vec.rs

```diff
@@ -7,6 +7,35 @@ use core::{
     ops, ptr, slice,
 };
 
+/// Workaround forbidden specialization of Drop
+pub trait VecDrop {
+    unsafe fn drop_with_len(&mut self, len: usize);
+}
+
+impl<T> VecDrop for [MaybeUninit<T>] {
+    unsafe fn drop_with_len(&mut self, _len: usize) {
+        // Case of a view, drop does nothing
+    }
+}
+
+impl<T, const N: usize> VecDrop for [MaybeUninit<T>; N] {
+    unsafe fn drop_with_len(&mut self, len: usize) {
+        // NOTE(unsafe) avoid bound checks in the slicing operation
```

Comment:
This only explains why unsafe code is used, not why it is safe. Also, the `unsafe` blocks in here are not needed anymore.

Reply:
Good point. Fixed in https://github.com/rust-embedded/heapless/pull/425/commits/fe3dd4d64af313533999e2014361fcd3c2db31cf